src/components/expression/expression-renderer.js

```diff
@@ -1338,59 +1318,121 @@ genotet.ExpressionRenderer.prototype.drawProfileLegend_ = function(profiles) {
 /**
  * Adds the expression profiles for the selected genes as line charts.
  * @param {number} geneIndex
+ * @param {string} geneName
  */
-genotet.ExpressionRenderer.prototype.addGeneProfile = function(geneIndex) {
+genotet.ExpressionRenderer.prototype.addGeneProfile = function(geneIndex,
+                                                               geneName) {
   var profile = new genotet.ExpressionRenderer.Profile({
-    geneName: this.data.matrix.geneNames[geneIndex],
+    geneName: geneName,
     row: geneIndex
   });
   this.data.profiles.push(profile);
-  this.drawGeneProfiles_();
+  this.drawGeneProfiles();
 };
 
 /**
- * Adds the expression profiles for the selected genes as line charts.
- * @param {number} geneIndex
+ * Removes the expression profiles for the selected genes as line charts.
+ * @param {string} geneName
  */
-genotet.ExpressionRenderer.prototype.removeGeneProfile = function(geneIndex) {
+genotet.ExpressionRenderer.prototype.removeGeneProfile = function(geneName) {
   var index = -1;
   for (var i = 0; i < this.data.profiles.length; i++) {
-    if (this.data.profiles[i].row == geneIndex) {
+    if (this.data.profiles[i].geneName == geneName) {
       index = i;
       break;
     }
   }
   this.data.profiles.splice(index, 1);
-  this.drawGeneProfiles_();
+  this.data.profiles.forEach(function(profile, i) {
+    profile.row = i;
+  }, this);
+
+  this.data.profile.geneNames.splice(index, 1);
+  this.data.profile.values.splice(index, 1);
+  if (this.data.profile.values.length == 0) {
+    this.data.profile.valueMin = -Infinity;
+    this.data.profile.valueMax = Infinity;
+  } else {
+    var profileValues = this.data.profile.values.reduce(function(prev, next) {
```

Comment:
Inefficient. You are only reducing to get min/max. However you can achieve that by a single scan.